packages/react-charting/UnitTests/HeatMapChartUT.test.tsx

```diff
@@ -0,0 +1,676 @@
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const label = instance._getFormattedLabelForXAxisDataPoint('');
+    expect(label).toEqual('');
+  });
+});
+
+runTest('_getFormattedLabelForYAxisDataPoint', () => {
+  test('Should return proper yAxis label for non empty string', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const label = instance._getFormattedLabelForYAxisDataPoint(yPoint[0]);
+    expect(label).toEqual('p1');
+  });
+
+  test('Should return proper yAxis label for empty data', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const label = instance._getFormattedLabelForYAxisDataPoint('');
+    expect(label).toEqual('');
+  });
+});
+
+runTest('_getStringFormattedNumber', () => {
+  test('Should return proper string formatted number for numeric value', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedNumber('1000');
+    expect(result).toEqual('1k');
+  });
+
+  test('Should return proper string formatted number with format string', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedNumber('10000', '.2~s');
+    expect(result).toEqual('10k');
+  });
+
+  test('Should return proper string formatted number for empty data', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedNumber('');
+    expect(result).toEqual('0');
+  });
+
+  test('Should return proper string formatted number for string value', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedNumber('Test1');
+    expect(result).toEqual('NaN');
+  });
+});
+
+runTest('_getStringFormattedDate', () => {
+  test('Should return proper string formatted date for date point', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedDate('100000000');
```

Comment:
This test case is interesting. How is this returning Jan/02

Reply:
While setting the time in milli seconds date is always considering January 1, 1970 GMT as current date.
    date.setTime(+point)